cdap-examples/DataCleansing/src/test/java/co/cask/cdap/examples/datacleansing/DataCleansingMapReduceTest.java

```diff
@@ -0,0 +1,179 @@
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URL;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests that a MapReduce job can incrementally process the partitions of a PartitionedFileSet, using a small sample of
+ * data with the DataCleansing MapReduce job.
+ */
+public class DataCleansingMapReduceTest extends TestBase {
+  private static final Set<String> RECORDS1 =
+    ImmutableSet.of("{\"pid\":223986723,\"name\":\"bob\",\"dob\":\"02-12-1983\",\"zip\":\"84125\"}",
+                    "{\"pid\":198637201,\"name\":\"timothy\",\"dob\":\"06-21-1995\",\"zip\":\"84125q\"}");
+  private static final Set<String> RECORDS2 =
+    ImmutableSet.of("{\"pid\":001058370,\"name\":\"jill\",\"dob\":\"12-12-1963\",\"zip\":\"84125\"}",
+                    "{\"pid\":000150018,\"name\":\"wendy\",\"dob\":\"06-19-1987\",\"zip\":\"84125\"}");
+  private static final Set<String> RECORDS3 =
+    ImmutableSet.of("{\"pid\":013587810,\"name\":\"john\",\"dob\":\"10-10-1991\",\"zip\":\"84125\"}",
+                    "{\"pid\":811638015,\"name\":\"samantha\",\"dob\":\"04-20-1965\",\"zip\":\"84125\"}");
+
+  private static final String schemaJson = DataCleansingMapReduce.SchemaMatchingFilter.DEFAULT_SCHEMA.toString();
+  private static final SimpleSchemaMatcher schemaMatcher =
+    new SimpleSchemaMatcher(DataCleansingMapReduce.SchemaMatchingFilter.DEFAULT_SCHEMA);
+
+  @Test
+  public void testPartitionConsuming() throws Exception {
+    ApplicationManager applicationManager = deployApplication(DataCleansing.class);
+
+    ServiceManager serviceManager = applicationManager.getServiceManager(DataCleansingService.NAME).start();
+    serviceManager.waitForStatus(true);
+    URL serviceURL = serviceManager.getServiceURL();
+
+    // write a set of records to one partition and run the DataCleansingMapReduce job on that one partition
+    createPartition(serviceURL, RECORDS1);
+
+    // before starting the MR, there are 0 invalid records, according to metrics
+    Assert.assertEquals(0, getInvalidDataCount());
+    Long now = System.currentTimeMillis();
+    ImmutableMap<String, String> args = ImmutableMap.of(DataCleansingMapReduce.OUTPUT_PARTITION_KEY, now.toString(),
+                                                        DataCleansingMapReduce.SCHEMA_KEY, schemaJson);
+    MapReduceManager mapReduceManager = applicationManager.getMapReduceManager(DataCleansingMapReduce.NAME).start(args);
+    mapReduceManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    Assert.assertEquals(filterInvalidRecords(RECORDS1), getCleanData(now));
```

Comment:
That's what I was originally doing (serving the file from a user service), but after a discussion with @terence and @sreevatsanraman , we came to the conclusion that we should try not to demonstrate doing that, because it would not scale if the data size is actually large (which a user may try this example with larger data).